### PR TITLE
Improve Asset Dropdown

### PR DESCRIPTION
**Pull request by blurpesec**
### Description
Changed asset dropdown a bit to be more performant.

##### Changes:
- Removed Typography from asset selector
- Aligned items in the selector a little better (fixes an issue with the blinking cursor in searchable Selector lacking padding-left).
- Removed displaying of asset icons when using large list of assets (Hardware Scanner & Request Assets).

image

### CODE REVIEW

File: src/features/PurchaseMembership/components/MembershipSelector.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { IMembershipConfig, MEMBERSHIP_CONFIG } from '../config';
 const SContainer = styled('div')`
   display: flex;
   flex-direction: row;
-  padding: 12px;
+  padding: 12px 12px 12px 0px;
```
Comment (FrederikBolding) [inline src/features/PurchaseMembership/components/MembershipSelector.tsx:15]: image
Membership looks off.

File: src/components/AssetSelector.test.tsx [hunk 2]
```diff
@@ -35,14 +36,14 @@ describe('AssetSelector', () => {
...
-    expect(screen.getByText(fAssets[0].ticker)).toBeInTheDocument();
+    expect(screen.getByText(`${fAssets[0].ticker} - ${fAssets[0].name}`)).toBeInTheDocument();
   });
 
   test('it is searchable by name', async () => {
     const props = Object.assign({}, defaultProps, { searchable: true });
     const { container } = getComponent(props);
     fireEvent.change(container.querySelector('input')!, { target: { value: fAssets[0].name } });
-    expect(screen.getAllByText(fAssets[0].name)).toHaveLength(2);
+    expect(screen.getAllByText(fAssets[0].name)).toHaveLength(1); //@todo: fixThis - It should detect Ethereum Classic
```
Comment (FrederikBolding) [inline src/components/AssetSelector.test.tsx:46]: Can we fix this now?

File: src/components/Selector.tsx [hunk 3]
```diff
@@ -153,9 +153,10 @@ const customStyles: Styles = {
     display: 'inline-block'
   }),
   // Allow the valueComponent to handle it's own padding when present.
+  // If input is present in the field, it takes up 4px.
   valueContainer: (styles, state) => ({
     ...styles,
-    ...(state && state.hasValue && { paddingLeft: 0 })
+    paddingLeft: state.hasValue ? '6px' : '10px'
```
Comment (FrederikBolding) [inline src/components/Selector.tsx:159]: Can you explain this change?

File: src/components/NetworkNodeDropdown.tsx [hunk 4]
```diff
@@ -20,7 +20,7 @@ const SContainer = styled.div`
 `;
 
 const SContainerValue = styled(SContainer)`
-  padding: ${SPACING.XS};
+  padding: ${SPACING.XS} ${SPACING.XS} ${SPACING.XS} 0px;
```
Comment (FrederikBolding) [inline src/components/NetworkNodeDropdown.tsx:23]: image

File: src/features/InteractWithContracts/components/FunctionDropdownItem.tsx [hunk 5]
```diff
@@ -27,7 +27,7 @@ const Sticker = styled.div<StickerProps>`
   background-color: ${(props) => (props.isRead ? COLORS.SUCCESS_GREEN : COLORS.BLUE_BRIGHT)};
   border-radius: 28px;
   color: ${COLORS.WHITE};
-  padding: 2px 8px;
+  padding: 2px 8px 2px 0px;
```
Comment (FrederikBolding) [inline src/features/InteractWithContracts/components/FunctionDropdownItem.tsx:30]: image
Badge is off

File: src/features/InteractWithContracts/components/fields/BooleanSelector.tsx [hunk 6]
```diff
@@ -34,7 +34,7 @@ const ContractDropdownItem = ({
   option: Contract;
   onSelect?(option: Contract): void;
 }) => (
-  <div style={{ padding: '12px 15px' }} onClick={() => onSelect && onSelect(option)}>
+  <div style={{ padding: '12px 15px 12px 0px' }} onClick={() => onSelect && onSelect(option)}>
```
Comment (FrederikBolding) [inline src/features/InteractWithContracts/components/fields/BooleanSelector.tsx:37]: image

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

**Comment by FrederikBolding** [comment]
[ch8049]

File: src/components/Selector.tsx (hunk 3 above)
Comment (blurpesec) [reply src/components/Selector.tsx:159]: Basically, selectors already have padding in them that they use for spacing of the searchable field's blinking cursor & text elements when searching. We were overriding this, resulting in the cursor being in a weird spot. old: 
image

new:
image

**Comment by LuckyLuker** [comment]
[ch8049]

**Review by FrederikBolding — DISMISSED** [review]

**Review by FrederikBolding — CHANGES_REQUESTED** [review]
image
Small regression following your rebase

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**